demo/clusterctl-examples/README.md

```diff
@@ -0,0 +1,149 @@
+# Table Of Contents
+
+- [Overview](#overview)
+  - [OS Image](#os_image)
+  - [Network Configurations](#network_configurations)
+  - [Network Access](#network_access)
+  - [User Accounts](#user_accounts)
+- [Prerequsites](#prerequisites)
+- [How-To](#how_to)
+
+# Overview <a name="overview" />
+
+This directory contains sample [clusterctl templates][clusterctl-templates] for
+both [cri-o][cri-o] and [rke2][rke2] clusters. The boostrapping and
+configuration of these clusters will be largely depended on the following:
+
+## OS Image <a name="os_image" />
+
+The type of OS image (i.e. SUSE-based, Ubuntu-based, etc) and what's on that
+image (i.e. packages pre-installed, package availability, networking setup,
+security setup, user accounts setup, etc) will determine what the cluster
+bootstrapping commands will look like.
+
+## Network Configurations <a name="network_configurations" />
+
+The network configuration is very much environment-specific. *The NIC names
+for the baremetal nodes must be consistent as they are sharing the same
+*Metal3DataTemplate*. For example, we can't have a NIC in one node named
+`eno1f0` while the other node have it as `eth0`.
+
+Furthermore, the configuration of IP pools, gateways, DNS, and routing table
+required indepth knowledge of infrastructure networking setup.
+
+## Network Access <a name="network_access" />
+
+Whether the baremetal provisioning network have public network egress or not
+(i.e. air-gapped environment) will also influence the cluster bootstrapping
+commands. For an air-gapped environment, any public endpoints will need to be
+updated to their respective internal endpoints.
+
+## User Accounts <a name="user_accounts" />
+
+The bootstrapping commands may also depended on certain user accounts (for
+management and monitoring purposes as an example).
+
+# Prerequsites <a name="prerequisites" />
+
+The samples were validated based on the following:
+
+1. Ubuntu Jimmy OS image.
+2. Two baremetal nodes, one for the control plane and the other for worker.
```

Comment:
Not sure if it belongs here (or if it's addressed elsewhere), but the bare-metal nodes need a network-connected BMC and TPM 2.0. In addition, I've not seen the spelling "baremetal", I've only seen it as "bare-metal" or "bare metal".

Reply:
I was wondering about the difference between 'baremetal', 'bare metal', and 'bare-metal' as well. Let's go with 'bare metal' for now. :-) I didn't enable TPM when validating the samples as I wasn't using the Elemental image.

Reply:
That's interesting. I didn't know that you only needed TPM when using the Elemental version of SLES. That might be interesting info to have somewhere.

Reply:
That's true. Ironic doesn't care about TPM, it's the Elemental installer that does.

Reply:
It's the "secure device onboarding" feature that drives this requirement. And it's the only way to uniquely identify a piece of (edge) hardware.

There might be other ways to uniquely identify a system on public clouds etc. but that's not Elemental's _current_ target market.